Source/ACE/Entity/Player.cs

```diff
@@ -1605,5 +1605,30 @@ public void ClearDelayedTeleport()
             DelayedTeleportTime = DateTime.MinValue;
             WaitingForDelayedTeleport = false;
         }
+
+        override public void GameLoopUpdate()
+        {
+            uint oldHealth = Health.Current;
+            uint oldStamina = Stamina.Current;
+            uint oldMana = Mana.Current;
+
+            base.GameLoopUpdate();
+
+            // If the game loop changed a vital -- send an update message to the client
+            if (Health.Current != oldHealth)
+            {
+                Session.Network.EnqueueSend(new GameMessagePrivateUpdateVital(Session, Enum.Ability.Health, Health));
+            }
+
+            if (Stamina.Current != oldStamina)
+            {
+                Session.Network.EnqueueSend(new GameMessagePrivateUpdateVital(Session, Enum.Ability.Stamina, Stamina));
+            }
+
+            if (Mana.Current != oldMana)
+            {
+                Session.Network.EnqueueSend(new GameMessagePrivateUpdateVital(Session, Enum.Ability.Mana, Mana));
+            }
+        }
     }
```

Comment:
Keep in mind that for vitals changing positive or negative in value (IE heal over time) the proper message is GameMessagePrivateUpdateAttribute2ndLevel, not GameMessagePrivateUpdateVital. I think we've recently discovered there are unique sequences associated each vital.